src/theme/Layout/hooks.ts

```diff
@@ -0,0 +1,77 @@
+import { trySafe, type Optional } from '@silverhand/essentials';
+import { useEffect, useMemo, useState } from 'react';
+
+import { defaultApiBaseProdUrl, defaultApiBaseDevUrl } from './constants';
+import { createDebugLogger, type DebugLogger } from './debug-logger';
+import { type GoogleOneTapConfig, googleOneTapConfigSchema } from './google-one-tap';
+import type { SiteConfig } from './types';
+
+export function useDebugLogger(siteConfig: SiteConfig): DebugLogger {
+  const isDebugMode = Boolean(siteConfig.customFields?.isDebuggingEnabled);
+
+  return useMemo(() => createDebugLogger(isDebugMode), [isDebugMode]);
+}
+
+export function useApiBaseUrl(siteConfig: SiteConfig): {
```

Comment:
You can call
```
const { siteConfig } = useDocusaurusContext();
```
in this hook, instead of passing the `siteConfig` through the properties. This makes the hook easier to call at component level, as it has less dependencies of other hooks.